core.ghactivity.php

```diff
@@ -1148,37 +1167,55 @@ public function find_post( $repo_name, $issue_number ) {
 
 
 	/**
-	 * Remote call to get label events for every monitored repo
+	 * Remote call to get all label events for every monitored repo
 	 *
 	 * @since 2.1.0
 	 *
 	 * @return null|array
 	 */
-	public function get_github_issue_events() {
+	public function get_all_github_issue_events() {
 		$response_body    = array();
 		$repos_to_monitor = $this->get_monitored_repos( 'names' );
 		if ( empty( $repos_to_monitor ) ) {
 			return $response_body;
 		}
 		foreach ( $repos_to_monitor as $repo_name ) {
-			$query_url = sprintf(
-				'https://api.github.com/repos/%1$s/issues/events?access_token=%2$s&per_page=100',
-				esc_html( $repo_name ),
-				$this->get_option( 'access_token' )
-			);
-
-			$data = wp_remote_get( esc_url_raw( $query_url ) );
-			if (
-				is_wp_error( $data )
-				|| 200 !== $data['response']['code']
-				|| empty( $data['body'] )
-			) {
-				return $response_body;
-			}
-			$single_response_body = json_decode( $data['body'] );
+			$single_response_body = $this->get_github_issue_events( $repo_name );
 			$response_body        = array_merge( $single_response_body, $response_body );
 		}
 		return $response_body;
 	}
+
+	/**
+	 * Remote call to get label events for specific repo and issue.
+	 *
+	 * @param string $repo_name name of the repo.
+	 * @param int    $issue_number issue number.
+	 *
+	 * @since 2.1.0
+	 *
+	 * @return null|array
+	 */
+	public function get_github_issue_events( $repo_name, $issue_number = null ) {
+		$response_body = array();
+		$query_url     = sprintf(
+			'https://api.github.com/repos/%1$s/issues%2$s/events?access_token=%3$s&per_page=100',
+			esc_html( $repo_name ),
+			$issue_number ? '/' . $issue_number : '',
```

Comment:
Maybe we should sanitize `$issue_number` here, to make sure our URL is always formed properly?